lisa/tools/copy_back.py

```diff
@@ -0,0 +1,47 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+import os
+from pathlib import PurePath
+
+from lisa.executable import Tool
+from lisa.tools.ls import Ls
+
+
+class CopyBack(Tool):
+    @property
+    def command(self) -> str:
+        return ""
+
+    @property
+    def can_install(self) -> bool:
+        return False
+
+    def copy_dir(
+        self,
+        src: PurePath,
+        dest: PurePath,
+        sudo: bool = False,
+        recur: bool = True,
+    ) -> None:
+        # create the destination directory if it doesn't exist
+        destination_dir = dest / src.name
+        if not os.path.exists(destination_dir):
+            os.makedirs(destination_dir)
+
+        # get list of files and folders in the source directory
+        _files = self.node.tools[Ls].list_files(str(src))
+        _dirs = self.node.tools[Ls].list_dir(str(src))
+        files = [PurePath(file) for file in _files]
+        dirs = [PurePath(dir) for dir in _dirs]
+
+        # copy files
+        for file in files:
+            self.node.shell.copy_back(file, destination_dir / file.name)
```

Comment:
We should find a way to support copy with sudo. Can you check if it's possible with the shell.copy_back? If not, we may have to implement it by cat or other tools.

Reply:
`spurshell.get` doesn't have sudo options. How about copying the folder into a temporary directory with `sudo cp`, changing ownership to current user, and then copying back the content?

Reply:
ok, we can take it as a workaround. Make sure the temp files are deleted, after copied back. It takes longer time to copy twice. We should avoid copying too many files back.